tests/components/velbus/test_config_flow.py

```diff
@@ -0,0 +1,62 @@
+"""Tests for the Velbus config flow."""
+from homeassistant import data_entry_flow
+from homeassistant.components.velbus import config_flow
+from homeassistant.const import CONF_PORT, CONF_NAME
+
+from tests.common import MockConfigEntry
+
+PORT_SERIAL = '/dev/ttyACME100'
+PORT_TCP = '127.0.1.0.1:3788'
+
+
+def init_config_flow(hass):
+    """Init a configuration flow."""
+    flow = config_flow.VelbusConfigFlow()
+    flow.hass = hass
+    return flow
+
+
+async def test_user(hass):
+    """Test user config."""
+    flow = init_config_flow(hass)
+
+    result = await flow.async_step_user()
+    assert result['type'] == data_entry_flow.RESULT_TYPE_FORM
+    assert result['step_id'] == 'user'
+
+    result = await flow.async_step_user({
+        CONF_NAME: 'Velbus Test Serial', CONF_PORT: PORT_SERIAL})
+    assert result['type'] == data_entry_flow.RESULT_TYPE_FORM
+    assert result['errors'] == {'port': 'connection_failed'}
```

Comment:
We should patch the velbus controller to test the successful flows.

Then add a test where we patch the controller to fail and raise.

We should never make I/O in the tests, so the controller needs to be patched in all cases where we test the connection.